src/mpid/common/genq/mpidu_genq_shmem_pool.c

```diff
@@ -87,7 +87,7 @@ int MPIDU_genq_shmem_pool_create_unsafe(uintptr_t cell_size, uintptr_t cells_per
         (MPIDU_genq_shmem_queue_u *) ((char *) pool_obj->slab + total_cells_size);
 
     rc = MPIDU_genq_shmem_queue_init(&pool_obj->free_queues[rank],
-                                     MPIDU_GENQ_SHMEM_QUEUE_TYPE__MPSC);
+                                     MPIDU_GENQ_SHMEM_QUEUE_TYPE__MPMC);
```

Comment:
I thought we going for `Multiple Producer Single Consumer`? Anyway, as I suggested, this need be separate PR from the memcpy utility PR. I'll spend more time later on this.

Reply:
As I mentioned in the commit message, I found that implementing SPMC was no cheaper than implementing MPMC. I might have missed something obvious here though. I can make it a separate PR, but it would be a requirement for this PR so I think they all have to go in together anyway. I find separating PRs that depend on each other into multiple PRs just makes it harder to review them.

Reply:
This discussion should move to https://github.com/pmodels/mpich/pull/5759.